modules/gcp/gke/variables.tf

```diff
@@ -0,0 +1,197 @@
+# Variables for Private Cluster
+####################
+variable "enable_private_endpoint" {
+  type        = bool
+  description = "Kubernetes cluster endpoint get private ip if you pass it as true"
+  default     = false
+}
+
+variable "enable_private_nodes" {
+  type        = bool
+  description = "Nodes inside the nodepool get's privateip only when we pass is as true. Since we are using private subnets we want all of our nodes to have priavate IP"
+  default     = false
+}
+
+/* Note: When you Pass "enable_private_endpoint" to true you must pass the value for below variable */
+variable "master_authorized_networks" {
+  type        = list(object({ cidr_block = string, display_name = string }))
+  description = "List of master authorized networks. If none are provided, disallow external access (except the cluster node IPs, which GKE automatically whitelists)."
+  default     = []
+}
+
+/* CIDR Block will be used by GKE and creats endpoint and peer with our network */
+variable "master_ipv4_cidr_block" {
+  type        = string
+  description = "(Beta) The IP range in CIDR notation to use for the hosted master network"
+  default     = "10.0.0.0/28"
+}
+
+
+/* Addons */
+variable "http_load_balancing" {
+  type        = bool
+  description = "Enable httpload balancer addon"
+  default     = false
+}
+
+
+variable "filestore_csi_driver" {
+  type        = bool
+  description = "The status of the Filestore CSI driver addon, which allows the usage of filestore instance as volumes"
+  default     = true
+}
+
+variable "enable_vertical_pod_autoscaling" {
+  type        = bool
+  description = "Vertical Pod Autoscaling automatically adjusts the resources of pods controlled by it"
+  default     = false
+}
+
+variable "horizontal_pod_autoscaling" {
+  type        = bool
+  description = "Enable horizontal pod autoscaling addon"
+  default     = true
+}
+
+
+variable "kubernetes_version" {
+  type        = string
+  description = "The Kubernetes version of the masters."
+  default     = "latest"
+}
+
+variable "initial_node_count" {
+  type        = number
+  description = "The number of nodes to create in this cluster's default node pool."
+  default     = 1
+}
+
+variable "cluster_resource_labels" {
+  type        = map(string)
+  description = "The GCE resource labels (a map of key/value pairs) to be applied to the cluster"
+  default     = {}
+}
+
+variable "node_pools" {
+  type        = list(map(string))
+  description = "List of maps containing node pools"
+
+  default = [
+    {
+      name = "default-node-pool"
+    },
+  ]
+}
+
+variable "remove_default_node_pool" {
+  type        = bool
+  description = "Remove default node pool while setting up the cluster"
+  default     = true
+}
+
+variable "node_pools_labels" {
+  type        = map(map(string))
+  description = "Map of maps containing node labels by node-pool name"
+
+  # Default is being set in variables_defaults.tf
+  default = {}
+}
+
+variable "node_pools_taints" {
```

Comment:
Since there are different label fields in module. Have a field called `default_labels` and these will be added to every resource that supports labels, apart from their own "resource" `labels` variable which would be `[resourceName]_labels`. So label assignment will look something like this:
```hcl
resource "node_pool" "my_node_pool" {
  labels = merge(var.default_labels, var.node_pool_labels)
}
```